app/src/main/java/com/example/trainingintive/navigators/Navigator.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.trainingintive.navigators
+
+import androidx.appcompat.app.AppCompatActivity
+import com.example.trainingintive.util.Event
+
+interface Navigator {
```

Comment:
It should be rather an abstract class because probably you don't want to repeat all of the declared methods (in this Navigator) in each of new navigator class